aperture.go

```diff
@@ -240,11 +240,44 @@ func getTLSConfig(serverName string, autoCert bool) (*tls.Config, error) {
 		log.Infof("Done generating TLS certificates")
 	}
 
-	certData, _, err := cert.LoadCert(tlsCertFile, tlsKeyFile)
+	certData, parsedCert, err := cert.LoadCert(tlsCertFile, tlsKeyFile)
 	if err != nil {
 		return nil, err
 	}
 
+	// If the certificate expired or it was outdated, delete it and the TLS
+	// key and generate a new pair.
+	if time.Now().After(parsedCert.NotAfter) {
```

Comment:
So the workflow is to wait until the certificates has expired and then restart the service? Sounds tricky to do without downtime.

Can't the certificate just never expire?

Reply:
This is the exact code that is used in `lnd`. But you are right, that doesn't make a lot of sense for a server that should have as little down time as possible.
No expiration is not possible, most TLS libraries will block a certificate that's valid for too long. But I chose a sane period that even the more restrictive libraries/OSes should allow and set the refresh check to about half that time. So as long as between year 1 and year 2 there is a restart, it should be fine.